pkg/addons/addons.go

```diff
@@ -570,3 +572,59 @@ func UpdateConfig(cc *config.ClusterConfig, enabled []string) {
 		}
 	}
 }
+
+// CheckPaused checks whether the cluster is paused before enable/disable an addon.
+func CheckPaused(profile string, val string) error {
+	klog.Info("checking whether the cluster is paused")
+
+	enable, err := strconv.ParseBool(val)
+	if err != nil {
+		return errors.Wrapf(err, "parsing bool: %s", val)
+	}
+
+	cc, err := config.Load(profile)
+	if err != nil {
+		return errors.Wrap(err, "loading profile")
+	}
+
+	api, err := machine.NewAPIClient()
+	if err != nil {
+		return errors.Wrap(err, "machine client")
+	}
+	defer api.Close()
+
+	cp, err := config.PrimaryControlPlane(cc)
+	if err != nil {
+		exit.Error(reason.GuestCpConfig, "Error getting primary control plane", err)
+	}
+
+	host, err := machine.LoadHost(api, config.MachineName(*cc, cp))
+	if err != nil {
+		exit.Error(reason.GuestLoadHost, "Error getting host", err)
```

Comment:
It's a bit odd that it switches between `return errors.Wrap` and `exit.Error`. The calling function will handle `exit.Error` for us so I think it makes sense to keep to just keep consistent with `return errors.Wrap`.

Reply:
Updated to use `errors.Wrap`.